packages/smart-contracts/test/contracts/BatchErc20Payments.test.ts

```diff
@@ -0,0 +1,400 @@
+      .to.emit(token, 'Transfer')
+      .to.emit(token, 'Transfer')
+      .to.emit(batch, 'TransferWithReferenceAndFee')
+      .withArgs(
+        tokenAddress,
+        payeeOne,
+        '20',
+        ethers.utils.keccak256(referenceExample1),
+        '1',
+        feeAddress,
+      )
+      .to.emit(batch, 'TransferWithReferenceAndFee')
+      .withArgs(
+        tokenAddress,
+        payeeTwo,
+        '30',
+        ethers.utils.keccak256(referenceExample2),
+        '2',
+        feeAddress,
+      )
+      .to.emit(batch, 'TransferWithReferenceAndFee')
+      .withArgs(
+        tokenAddress,
+        payeeThree,
+        '40',
+        ethers.utils.keccak256(referenceExample3),
+        '3',
+        feeAddress,
+      )
+      .to.emit(batch, 'TransferWithReferenceAndFee')
+      .withArgs(
+        tokenAddress,
+        payeeFour,
+        '50',
+        ethers.utils.keccak256(referenceExample4),
+        '4',
+        feeAddress,
+      );
+
+    afterERC20Balance = await token.connect(owner).balanceOf(payeeTwo);
+    await expect(afterERC20Balance).to.be.equal(beforeERC20Balance.add(30));
+  });
+
+  it('Should execute a batch payments of a ERC20 to four accounts', async function () {
+    beforeERC20Balance = await token.connect(owner).balanceOf(payeeThree);
+    await expect(
+      batch
+        .connect(spender)
+        .batchOrphanERC20Payments(
+          tokenAddress,
+          [payeeOne, payeeTwo, payeeThree, payeeFour],
+          [20, 30, 40, 50],
+        ),
+    )
+      .to.emit(token, 'Transfer')
+      .to.emit(token, 'Transfer')
+      .to.emit(token, 'Transfer')
+      .to.emit(token, 'Transfer');
+
+    afterERC20Balance = await token.connect(owner).balanceOf(payeeThree);
+    expect(afterERC20Balance).to.be.equal(beforeERC20Balance.add(40));
+  });
+
+  it('Should pay 4 ERC20 payments on 2 tokens', async function () {
+    beforeERC20Balance = await token.balanceOf(payeeTwo);
+    beforeERC20Balance1 = await token1.balanceOf(payeeTwo);
+
+    nbTxs = 4;
+    let [
+      tokenAddresses,
+      recipients,
+      amounts,
+      paymentReferences,
+      feeAmounts,
+    ] = getInputsBatchPayments(nbTxs, tokenAddress, payeeTwo, amount, referenceExample1, feeAmount);
+
+    tokenAddresses[2] = token1Address;
+    tokenAddresses[3] = token1Address;
+
+    tx = await batch
+      .connect(spender2)
+      .batchERC20PaymentsMultiTokensWithReferenceAndFee(
+        tokenAddresses,
+        recipients,
+        amounts,
+        paymentReferences,
+        feeAmounts,
+        feeAddress,
+      );
+    await tx.wait();
+
+    afterERC20Balance = await token.balanceOf(payeeTwo);
+    expect(afterERC20Balance).to.be.equal(beforeERC20Balance.add(amount * 2));
+
+    afterERC20Balance1 = await token1.balanceOf(payeeTwo);
+    expect(afterERC20Balance1).to.be.equal(beforeERC20Balance1.add(amount * 2));
+  });
+
+  it('GAS EVALUATION: Should pay multiple ERC20 payments', async function () {
```

Comment:
For the sake of a readable code, please nest the gas evaluation related unit tests within a describe section

Reply:
I xdeleted the loop...